tests/acct_mgt_test.py

```diff
@@ -0,0 +1,353 @@
+        stderr=subprocess.STDOUT,
+        check=False,
+    )
+    if result.returncode == 0:
+        result_json = json.loads(result.stdout.decode("utf-8"))
+        print(result_json["status"]["phase"])
+        return result_json["status"]["phase"]
+    print("None")
+    return None
+
+
+#
+def wait_while(project, pod_name, statuses, time_out=300):
+    """
+    This basically waits while a pod is in a set of statuses and possibly times out
+    It is used to wait for openshift to do something with the pod, as in, deleting
+    or starting a pod.
+
+    pass in the following parameter:
+       project:  the namespace of the pod
+       pod_name: the name of the pod
+       statuses: array of statuses
+
+
+    """
+    time_left = time_out
+    time_interval = 5
+    time.sleep(time_interval)
+    status = get_pod_status(project, pod_name)
+    while status in statuses and time_left > 0:
+        time.sleep(time_interval)
+        time_left = time_left - time_interval
+        status = get_pod_status(project, pod_name)
+
+    if status in statuses:
+        return False
+    return True
+
+
+def wait_until_done(oc_cmd, finished_pattern, time_out=30, decrement=5):
+    """
+    This wait while an oc command is running, looking for a pattern that
+    indicates it is finished.
+    """
+    pattern = re.compile(finished_pattern)
+    done = False
+    oc_array = oc_cmd.split(" ")
+    matched_line = ""
+    while time_out > 0 and not done:
+        time.sleep(5)
+        time_out = time_out - decrement
+        result = subprocess.run(
+            oc_array,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            check=False,
+        )
+        line_list = result.stdout.decode("utf-8").split("\n")
+        for line in line_list:
+            if pattern.match(line):
+                matched_line = line
+                done = True
+    if pattern.match(matched_line):
+        return True
+    return False
+
+
+def compare_results(result, pattern):
+    """This compares the result of a subprocess (usually curl) call  with a pattern"""
+    if result is not None:
+        pattern = re.compile(pattern)
+        line_array = result.stdout.decode("utf-8").split("\n")
+        for line in line_array:
+            if pattern.match(line):
+                return True
+    return False
+
+
+def oc_resource_exist(resource, kind, name, project=None):
+    """This uses oc to determine if an openshift resource exists"""
+    result = None
+    if project is None:
+        result = subprocess.run(
+            ["oc", "-o", "json", "get", resource, name],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            check=False,
+        )
+    else:
+        result = subprocess.run(
+            ["oc", "-o", "json", "-n", project, "get", resource, name],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            check=False,
+        )
+    if result.returncode == 0:
```

Comment:
Isn't the return code sufficient? It seems like you could just:

```
return result.returncode == 0
```

Reply:
for context:

    if result.returncode == 0:
        result_json = json.loads(result.stdout.decode("utf-8"))
        return result_json["status"]["phase"]
    return None

Here I am not returning the "resutl.returncode==0", that is either True or False, I am returning a specific field in the json output as you suggested before and as you suggested in a previous comment above.

When somethings are changing status in openshift (pods for example) they go though a series of different statuses, each one can be captured by oc on multiple calls.  It is in the testing code because I found it useful to tell when openshift had finished deployment so that I could start running the tests.  I am considering making use of something similar to wait for the quota change, since that also has some weird behavior.

Reply:
And yes I can probably do the equivalent of returning the return code, and I am in the process of removing the objectionable code as I am typing this.